processor/k8sattributesprocessor/e2e_test.go

```diff
@@ -50,139 +71,376 @@ func newExpectedValue(mode int, value string) *expectedValue {
 	}
 }
 
-func TestTraceE2E(t *testing.T) {
+// TestE2E tests the k8s attributes processor with a real k8s cluster.
+// The test requires a prebuilt otelcontribcol image uploaded to a kind k8s cluster defined in
+// `/tmp/kube-config-otelcol-e2e-testing`. Run the following command prior to running the test locally:
+//
+//	kind create cluster --kubeconfig=/tmp/kube-config-otelcol-e2e-testing
+//	make docker-otelcontribcol
+//	KUBECONFIG=/tmp/kube-config-otelcol-e2e-testing kind load docker-image otelcontribcol:latest
+func TestE2E(t *testing.T) {
+	kubeConfig, err := clientcmd.BuildConfigFromFlags("", testKubeConfig)
+	require.NoError(t, err)
+	dynamicClient, err := dynamic.NewForConfig(kubeConfig)
+	require.NoError(t, err)
+
+	testID := uuid.NewString()[:8]
+	collectorObjs, err := createCollectorObjects(dynamicClient, testID)
```

Comment:
Do we need to wait for collector entering into running states? In case of some invalid config or other reason may cause collector crash.

Reply:
Sounds good. Added the a step to wait for collector